lib/Mir.ml

```diff
@@ -94,7 +94,7 @@ and 'e expr =
 type fun_arg_decl = (autodifftype * string * unsizedtype) list
 
 and ('e, 's) statement =
-  | Assignment of 'e * 'e
+  | Assignment of string * 'e index list * 'e
```

Comment:
It might be useful to name this type, since it's also used in reaching definitions and dependency analysis